src/lib/rnp.cpp

```diff
@@ -5092,8 +5092,17 @@ try {
     if (password && (ret = rnp_op_generate_set_protection_password(subop, password))) {
         goto done;
     }
-    if ((ret = rnp_op_generate_add_usage(subop, "encrypt"))) {
-        goto done;
+    if (pgp_pk_alg_capabilities(subop->crypto.key_alg) & PGP_KF_ENCRYPT) {
+        if ((ret = rnp_op_generate_add_usage(subop, "encrypt"))) {
+            goto done;
+        }
+    } else {
+        if ((ret = rnp_op_generate_add_usage(subop, "sign"))) {
+            goto done;
+        }
+        if ((ret = rnp_op_generate_add_usage(subop, "certify"))) {
```

Comment:
@antonsviridenko 
Not sure whether we should add certification flag by default, as these are different operations. Is it something which GnuPG does for signing subkey?

Reply:
Yes, right, "certify" is set only for primary keys, signing subkeys have signing flag only in GnuPG.